File: Freedom/聚合数据/联系我们/JuheContectUSViewController.m
//  JuheContectUSViewController.m
//  Created by Super on 16/9/5.
//  Copyright © 2016年 Super. All rights reserved.
//
#import "JuheContectUSViewController.h"
#import "JuhePublicViewController.h"
#import "JuheMessageViewController.h"
#import "JuheAboutUSViewController.h"
#import "JuheChartViewController.h"
@interface JuheContectUSViewCell:BaseCollectionViewCell{
    NSMutableDictionary *thumbnailCache;
}
-(void)setCollectionDataWithDic:(NSDictionary*)dict;
@end
@implementation JuheContectUSViewCell
-(void)initUI{
    [super initUI];
    self.icon.frame = CGRectMake(0, 0, APPW*2/3, 120);
    self.title.frame = CGRectMake(0, YH(self.icon), W(self.icon), 20);
    self.title.textAlignment = NSTextAlignmentCenter;
    [self addSubviews:self.icon,self.title,nil];
}
-(void)setCollectionDataWithDic:(NSDictionary*)dict{
    
    self.title.text = dict[@"name"];
    __block UIImage *imageProduct = [thumbnailCache objectForKey:dict[@"pic"]];
    if(imageProduct){
        self.icon.image = imageProduct;
    }else{
        dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0ul);
        dispatch_async(queue, ^{
            UIImage *image = [UIImage imageNamed:dict[@"pic"]];
            dispatch_async(dispatch_get_main_queue(), ^{
                self.icon.image = image;
                [thumbnailCache setValue:image forKey:dict[@"pic"]];
            });
        });
    }
}
@end
@implementation JuheContectUSViewController
-(void)viewDidLoad{
    [super viewDidLoad];
    self.title = @"联系我们";
    BaseCollectionViewLayout *layout = [BaseCollectionViewLayout sharedFlowlayoutWithCellSize:CGSizeMake(APPW*2/3, 140) groupInset:UIEdgeInsetsMake(10, 10, 0, 10) itemSpace:50 linespace:10];
    self.collectionView = [[BaseCollectionView alloc]initWithFrame:CGRectMake(0, 0, APPW, APPH-TabBarH) collectionViewLayout:layout];
   
    self.collectionView.dataArray = [NSMutableArray arrayWithObjects:@{@"name":@"进入公众号",@"pic":@"juheintopublic"},@{@"name":@"查看历史消息",@"pic":@"juhelookhistory"},@{@"name":@"关于我们",@"pic":@"juheaboutus"},@{@"name":@"聊天",@"pic":PuserLogo}, nil];
    [self fillTheCollectionViewDataWithCanMove:NO sectionN:1 itemN:4 itemName:@"JuheContectUSViewCell"];
    self.collectionView.delegate = self;
    self.collectionView.dataSource = self;
    [self.view addSubview:self.collectionView];
}
-(void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath{
    if(indexPath.row==0){
        [self pushController:[JuhePublicViewController class] withInfo:nil withTitle:self.collectionView.dataArray[indexPath.row][@"name"]];
    
    }else if(indexPath.row==1){
        [self pushController:[JuheMessageViewController class] withInfo:nil withTitle:self.collectionView.dataArray[indexPath.row][@"name"]];
    }else if(indexPath.row==2){
        [self pushController:[JuheAboutUSViewController class] withInfo:nil withTitle:self.collectionView.dataArray[indexPath.row][@"name"]];
    }else{
        [self pushController:[JuheChartViewController class] withInfo:nil withTitle:self.collectionView.dataArray[indexPath.row][@"name"]];
    }
}
@end

File: Freedom/聚合数据/联系我们/JuheContectUSViewController.swift
//
//  JuheContectUSViewController.swift
//  Freedom
//
import UIKit
import BaseFile
import XExtension
import XCarryOn
class JuheContectUSViewCell:BaseCollectionViewCell{
    override func initUI() {
        self.icon = UIImageView(frame: CGRect(x:0, y:0, width:APPW*2/3, height:120))
        self.title = UILabel(frame: CGRect(x:0, y:YH( self.icon), width:self.icon.width(), height: 20))
        self.title.textAlignment = .center
        self.addSubviews([self.title,self.icon])
        self.title.text = "name"
        self.icon.image = UIImage(named:"taobaomini2")
    }
}
class JuheContectUSViewController: JuheBaseViewController,UICollectionViewDelegate,UICollectionViewDataSource {
    override func viewDidLoad() {
        super.viewDidLoad()
    self.title = "联系我们";
    let layout = UICollectionViewFlowLayout()
    layout.itemSize = CGSize(width: (APPW*2)/3, height:140);
    layout.sectionInset = UIEdgeInsets(top: 10, left: 10, bottom: 0, right: 10)
    layout.minimumInteritemSpacing = 50;
    layout.minimumLineSpacing = 10;
        self.collectionView = BaseCollectionView(frame: CGRect(x: 0, y: 0, width: APPW, height: APPH-TopHeight), collectionViewLayout: layout)
        
    self.collectionView.dataArray = NSMutableArray(array: ["进入公众号","查看历史消息","关于我们","聊天"])
    self.collectionView.delegate = self;
    self.collectionView.dataSource = self;
        self.collectionView.register(JuheContectUSViewCell.self, forCellWithReuseIdentifier: JuheContectUSViewCell.identifier())
    view.addSubview(collectionView)
    }
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return self.collectionView.dataArray.count
    }
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: JuheContectUSViewCell.identifier(), for: indexPath) as? JuheContectUSViewCell
        
        return cell!
    }
    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
        if indexPath.row==0{
            _ = self.push(JuhePublicViewController(), withInfo: "", withTitle: "")
        }else if indexPath.row==1{
            _ = self.push(JuheMessageViewController(), withInfo: "", withTitle: "")
        }else if indexPath.row==2{
            _ = self.push(JuheAboutUSViewController(), withInfo: "", withTitle: "")
        }else{
            _ = self.push(JuheChartViewController(), withInfo: "", withTitle: "")
        }
    }
}
